src/data/nat/factorization.lean

```diff
@@ -119,6 +120,32 @@ begin
     simp [prod_insert hxT, sum_insert hxT, ←IH, factorization_mul (hS x hxS) hT] }
 end
 
+/-! ### Bijection between pnats and finsupps `ℕ →₀ ℕ` with support on the primes -/
+
+/-- Any finsupp `f : ℕ →₀ ℕ` whose support is in the primes is equal to the factorization of
+the product `∏ (a : ℕ) in f.support, a ^ f a`. -/
+lemma factorization_prod_pow_inv {f : ℕ →₀ ℕ} (hf : ∀ (p : ℕ), p ∈ f.support → prime p) :
```

Comment:
Why the `inv` in the name? I would call it something like `factorization_prod_pow_eq_self`.

Reply:
The `inv` was intended to suggest that `factorization` is an inverse operation `ℕ+ → (ℕ →₀ ℕ)` to the process `f ↦ f.prod pow`.

We already have `factorization_prod_pow_eq_self {n : ℕ} (hn : n ≠ 0) : n.factorization.prod pow = n` which says that the composition in the other direction is the identity.  So perhaps `prod_pow_factorization_eq_self` for this lemma?  (Or swap the names the other way around?)

Is there a mathlib convention for naming lemmas that say that some function `f` is the post-inverse of `g`?

Reply:
Well, then it should be `prod_pow_factorization_eq_self` I guess. If you want to talk about left/right inverses you should state the lemma using `comp`, without any argument, but I think here it's better to have it like it already is.